src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let res = initialValue;
+    let start = 0;
+
+    if (arguments.length < 2) {
```

Comment:
```suggestion
    if (initialValue === undefined) {
```

your version works well but is better if your condition strict